doc/rls_mode.md

```diff
@@ -14,39 +14,92 @@ The type of the parameter is an object with the following fields:
 
 By default, it is `null`.
 
-### Debugging
+## Setting up
 
-There is an output channel named "Rust Language Server" which is used to show messages from RLS.
+First of all, you have to download the RLS sourcs:
 
-To open it, perform the following steps:
+```bash
+git clone https://github.com/rust-lang-nursery/rls
+```
 
-* Click "View" on the menu
-* Click "Output" on submenu
-* Click on the listbox which is to the right of the shown panel
-* Choose "Rust Language Server"
+Depending on whether you have rustup or not, there are different ways you can set up this plugin.
+
+* [Setting up without rustup](#without-rustup)
+* [Setting up with rustup](#with-rustup)
+
+#### Without rustup
 
-For making RLS print more data, refer the "Debug RLS" section below.
+**Note:** You should only do this if you do not have rustup because otherwise rustup will not work anymore.
 
-### Examples
+After you have cloned the sources, you need to download the latest nightly compiler. See the [Building section of the Rust repository](https://github.com/rust-lang/rust#building-from-source) for how to do this.
 
-#### RLS is installed
+You can now install the Rust Language Server globally with
+
+```bash
+cargo install
+```
+
+and set `"executable"` to `"rls"`:
 
 ```json
 "rust.rls": {
     "executable": "rls"
 }
 ```
 
-#### Source code of RLS is available
+--
+
+If you don't want to have it installed you can also run it from sources:
+
+```json
+"rust.rls": {
+    "executable": "cargo",
+    "args": ["run", "--manifest-path=/path/to/rls/Cargo.toml", "--release"]
+}
+```
+
+#### With rustup
+
+Make sure you do have [rustup](https://github.com/rust-lang-nursery/rustup.rs) with nightly toolchain.
```

Comment:
@KalitaAlexey Yes, it is. I used it because the section says "With rustup" and most people following that section will already have it. If it should only be described how to do that with rustup, I'll change that anyway.